modules/sigprofiler/1.0/sigprofiler.smk

```diff
@@ -0,0 +1,215 @@
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["sigprofiler"]`
+CFG = op.setup_module(
+    name = "sigprofiler",
+    version = "1.0",
+    subdirectories = ["inputs", "estimate", "extract", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _install_sigprofiler_genome,
+    _sigprofiler_input_maf,
+    _sigprofiler_output_tsv,
+    _sigprofiler_all,
+
+
+##### FUNCTIONS #####
+
+def get_dir(wildcards):
+    if "ID" in wildcards.type:
+        topdir = 'ID'
+    elif "SBS" in wildcards.type:
+        topdir = 'SBS'
+    elif "DBS" in wildcards.type:
+        topdir = 'DBS'
+
+    cc = config["lcr-modules"]["sigprofiler"]
+    mat = cc["dirs"]["inputs"]+f"matrices/{wildcards.seq_type}--{wildcards.genome_build}/{wildcards.sample_set}/output/{topdir}/{wildcards.sample_set}.{wildcards.type}.all"
+    ret = {'script' : cc["inputs"]["extractor"], 'mat' : mat}
+    return(ret)
+
+max_sigs = {
+    "SBS6"    : 10,
+    "SBS18"   : 10,
+    "SBS24"   : 20,
+    "SBS96"   : 20,
+    "SBS288"  : 25,
+    "SBS384"  : 25,
+    "SBS4608" : 30,
+    "SBS1536" : 30,
+    "SBS6144" : 35,
+    "ID28"    : 5,
+    "ID83"    : 5,
+    "ID96"    : 10,
+    "ID332"   : 10,
+    "ID415"   : 15,
+    "ID8628"  : 20,
+    "DBS78"   : 15,
+    "DBS150"  : 15,
+    "DBS186"  : 15,
+    "DBS1248" : 20,
+    "DBS2400" : 20,
+    "DBS2976" : 20
+}
+
+##### RULES #####
+
+rule _install_sigprofiler_genome:
+    output:
+        complete = CFG["dirs"]["inputs"] + "sigprofiler_prereqs/{genome_build}.installed"
+    params:
+        ref = lambda w: {"grch37":"GRCh37", "hg19":"GRCh37",
+                         "grch38": "GRCh38", "hg38": "GRCh38"}[w.genome_build]
+    conda: CFG["conda_envs"]["sigprofiler"]
+    shell:
+        op.as_one_line("""
+        python -c 'from SigProfilerMatrixGenerator import install as genInstall;
+        genInstall.install("{params.ref}", rsync = False, bash = True)'
+            &&
+        touch {output.complete}
+        """)
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _sigprofiler_input_maf:
+    input:
+        maf = CFG["inputs"]["maf"]
+    output:
+        maf = CFG["dirs"]["inputs"] + "matrices/{seq_type}--{genome_build}/{sample_set}/{sample_set}.maf"
+    run:
+        op.relative_symlink(input.maf, output.maf)
+
+
```

Comment:
Explain what this rule is actually doing with a comment at the top

Reply:
Added rule comments